src/comm/TCPSocket.cc

```diff
@@ -104,12 +106,14 @@ bool TCPSocket::connect(const Address& address)
 std::unique_ptr< TCPSocket > TCPSocket::create()
 {
     int tcp_socket = ::socket(AF_INET, SOCK_STREAM, 0);
+    sockaddr_in source;
+    memset(&source, 0, sizeof(_source));
```

Comment:
Yikes, this `memset` is not a good code smell.  Does `sockaddr_in` not initialize itself properly?

However we initialize `sockaddr_in`, I feel like it should happen in `TCPSocket` ctor instead of being the caller's responsibility.

Reply:
sockaddr_in is pure C, so my assumption was that you treat it like a C struct.

Reply:
The way `TCPSocket` currently works is that there is only 1 constructor - it's internal and so it always is going to pass a valid `_source`. The design of the class is essentially a base class which is kind of a factory ( valid objects are created by accept or create ) and create is done the way it is because of how `ConnClient` utilizes it and generating an actually valid socket only after `connect` is called.

Reply:
>The way `TCPSocket` currently works is that there is only 1 constructor

Why is it important for `TCPSocket` to have a single ctor? Por que no los dos? 

It seems to me if we're adding a new member to `TCPSocket`, shouldn't it be that class's responsibility to initialize it correctly? That strikes me as an implementation detail of `TCPSocket` that this method shouldn't have to care about.

Reply:
Right - and the way it is initialized properly is by requiring it to be passed in, just like the fd is required.